lib/nx_gzlib.c

```diff
@@ -162,6 +162,49 @@ gzFile nx_gzopen_(const char* path, int fd, const char *mode)
 	return file;
 }
 
+int nx_gzread(gzFile file, void *buf, unsigned len)
+{
+	gz_statep state = (gz_statep) file;
+	z_streamp stream = &(state->strm);
+	int ret, err;
+	const uInt max = len > 10? 10 : 1;
+	uLong last_total_out = stream->total_out;
+	unsigned char* next_in = malloc(sizeof(char)*max);
+
+	if (next_in == NULL)
+		return Z_MEM_ERROR;
+
+	stream->next_out = buf;
+	stream->avail_out = len;
+	do {
+		ret = read(state->fd, next_in, sizeof(char)*max);
+		stream->next_in = next_in;
+		if (ret == 0){
+			/* Flush all at the end of the file. */
+			err = nx_inflate(stream, Z_FINISH);
+			break;
+		}
+
+		stream->avail_in = ret;
+		err = nx_inflate(stream, Z_NO_FLUSH);
+
+		if (err != Z_OK && err != Z_STREAM_END){
+			state->err = err;
+			return 0;
```

Comment:
Returning here will cause a memory leak for `next_in`.

Reply:
Fixed by freeing it there.